stone/target/python_rsrc/stone_serializers.py

```diff
@@ -853,6 +855,73 @@ def _make_stone_friendly(
         alias_validators[data_type](ret)
     return ret
 
+# Adapted from:
+# http://code.activestate.com/recipes/306860-proleptic-gregorian-dates-and-strftime-before-1900/
+# Remove the unsupposed "%s" command. But don't do it if there's an odd
+# number of %s before the s because those are all escaped. Can't simply
+# remove the s because the result of %sY should be %Y if %s isn't
+# supported, not the 4 digit year.
+_ILLEGAL_S = re.compile(r'((^|[^%])(%%)*%s)')
+
+def _findall(text, substr):
+    # Also finds overlaps
+    sites = []
+    i = 0
+
+    while 1:
+        j = text.find(substr, i)
+
+        if j == -1:
+            break
+
+        sites.append(j)
+        i = j + 1
+
+    return sites
+
+# Every 28 years the calendar repeats, except through century leap years
+# where it's 6 years. But only if you're using the Gregorian calendar. ;)
+def _strftime(dt, fmt):
+    try:
+        return dt.strftime(fmt)
+    except ValueError:
+        if six.PY3 or dt.year > 1900:
```

Comment:
@braincore, I'm sorry I misunderstood your previous comment. Is this what you were referring to? Basically, this tries to use the library function first. If we're executing in Python 3, or if the year is valid, then re-`raise` any caught `ValueError`. Otherwise, we're in Python 2 and the year is something we can compensate for.

If I was extra paranoid, I might try to look at the error string of the `ValueError` just to make sure it wasn't due to something else (even though the year is less than 1900).